### ISSUE TITLE
time.h issues when cross-compiling

### ISSUE THREAD

**Issue by kost** [nmap/nmap#442]
When cross compiling Nmap 7.12 release on Android, getting following errors:

```
arm-linux-androideabi-g++ -c -I./liblinear -I./liblua -I./libdnet-stripped/include -I./libpcre  -I/sdcard/opt-arm/openssl-1.0.2h/include -I./libpcap -I./nbase -I./nsock/include -DHAVE_CONFIG_H -DNMAP_NAME=\"Nmap\" -DNMAP_URL=\"https://nmap.org\" -DNMAP_PLATFORM=\"arm-unknown-linux-androideabi\" -DNMAPDATADIR=\"/sdcard/opt/nmap-7.12/share/nmap\" -D_FORTIFY_SOURCE=2 -fvisibility=default -fPIE -Wall -fno-strict-aliasing   nmap_error.cc -o nmap_error.o
nmap_error.cc: In function 'void fatal(const char*, ...)':
nmap_error.cc:145:20: error: 'time' was not declared in this scope
   timep = time(NULL);
                    ^
nmap_error.cc: In function 'void pfatal(const char*, ...)':
nmap_error.cc:215:20: error: 'time' was not declared in this scope
   timep = time(NULL);
                    ^
make[1]: *** [nmap_error.o] Error 1
make[1]: Leaving directory `/data/work/nmapand/nmap-7.12'
make: *** [compile] Error 2
```

Problem is that time.h is not included in some of the needed files. 

Quick and dirty fix for these problems in Nmap:
https://github.com/kost/nmap-android/blob/68142c7e35302a07e4c8dc854e89d8f21df3c17b/android/patches/60-fix-time-h.diff

I guess you want to make it work on more general level than this.

**Comment by dmiller-nmap** [comment]
@kost Can you tell me (from `config.h`) whether `TIME_WITH_SYS_TIME` or `HAVE_SYS_TIME_H` are defined? This section in `service_scan.cc` should result in an appropriate header being included, for instance:

```c++
#if TIME_WITH_SYS_TIME
# include <sys/time.h>
# include <time.h>                                                                              
#else                                                                                           
# if HAVE_SYS_TIME_H
#  include <sys/time.h>
# else
#  include <time.h>
# endif                                                                                         
#endif
```

I think maybe we should focus on improving our configure-time detection rather than just adding `#include <time.h>` everywhere.

**Comment by dmiller-nmap** [comment]
@kost Do you have an update on this? Is the cross-compile build still broken without your patch? And can you show the output of `grep SYS_TIME nmap_config.h`? I'd like to get this fixed.

**Comment by dmiller-nmap** [comment]
Patch is similar to PR #1317, closed in ab79382f932
